setup.py

```diff
@@ -53,6 +53,8 @@
             'numpy = intake.source.npy:NPySource',
             'ndzarr = intake.source.zarr:ZarrArraySource',
             'zarr_cat = intake.catalog.zarr:ZarrGroupCatalog',
+            'alias = intake.source.derived:AliasSource',
+            'derived = intake.source.derived:DerivedSource'
```

Comment:
Agreed, I think it should be removed from the registry. In fact, since this is somewhat beta, all of the derived classes (except Alias, which already existed) should probably be referred to by their full dotted name for now.

Reply:
So for now, I'd be fine with removing the `derived` from the registry and suggest to merge the PR, so that we can get more experience. But do you already have an idea if in future people should be using the `DerivedSource` directly or if we'll use a subclass also if it is only for selecting the `first_discoverable`?

Reply:
No, I don't expect anyone to use DerivedSource directly - they would subclass, or use one of the subclasses that we provide.